src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,73 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.excepction.UserRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int VALID_AGE = 18;
+    private static final int MIN_LOGIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserIsNull(user);
+        isAgeNull(user);
+        isAgeValid(user);
+        checkLoginIsNull(user);
+        checkLoginIsValid(user);
+        isPasswordNull(user);
+        checkPasswordIsValid(user);
+        checkIfUserExists(user);
+        return storageDao.add(user);
+    }
+
+    private void checkUserIsNull(User user) {
```

Comment:
```suggestion
    private void verifyUserIsNotNull(User user) {
```
please fix in all places